LLAP/llap.sh

```diff
@@ -0,0 +1,438 @@
+#!/bin/bash
+
+#    Copyright 2021 Google, Inc.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+
+set -euxo pipefail
+
+##install xml modifiction tool....
+
+apt-get install -y xmlstarlet
+
+readonly NOT_SUPPORTED_MESSAGE="LLAP initialization action is not supported on Dataproc ${DATAPROC_VERSION}."
+[[ $DATAPROC_VERSION != 2.* ]] && echo "$NOT_SUPPORTED_MESSAGE" && exit 1
+
+# Variables for running this script
+readonly ROLE="$(/usr/share/google/get_metadata_value attributes/dataproc-role)"
+readonly LLAP_MASTER_FQDN="$(/usr/share/google/get_metadata_value attributes/dataproc-master)"
+readonly WORKER_NODE_COUNT=$(/usr/share/google/get_metadata_value attributes/dataproc-worker-count)
+readonly NODE_MANAGER_vCPU=$(xmlstarlet sel -t -v '/configuration/property[name = "yarn.nodemanager.resource.cpu-vcores"]/value' -nl /etc/hadoop/conf/yarn-site.xml)
+readonly NODE_MANAGER_MEMORY=$(xmlstarlet sel -t -v '/configuration/property[name = "yarn.nodemanager.resource.memory-mb"]/value' -nl /etc/hadoop/conf/yarn-site.xml)
+readonly YARN_MAX_CONTAINER_MEMORY=$(xmlstarlet sel -t -v '/configuration/property[name = "yarn.scheduler.maximum-allocation-mb"]/value' -nl /etc/hadoop/conf/yarn-site.xml)
+readonly LAST_CHAR_MASTER=${LLAP_MASTER_FQDN: -1}
+readonly HAS_SSD=$(/usr/share/google/get_metadata_value attributes/ssd)
+
+###check to see if HA or not
+if [[ $LAST_CHAR_MASTER == 'm' ]];then
```

Comment:
You can do HA check like this, https://github.com/GoogleCloudDataproc/initialization-actions/blob/master/zookeeper/zookeeper.sh#L68